presto-main/src/main/java/com/facebook/presto/operator/HttpTaskResultClient.java

```diff
@@ -69,467 +51,65 @@
-            Ticker ticker,
-            Executor pageBufferClientCallbackExecutor)
+    public HttpTaskResultClient(HttpClient httpClient, URI location)
     {
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
-        this.acknowledgePages = acknowledgePages;
         this.location = requireNonNull(location, "location is null");
-        this.clientCallback = requireNonNull(clientCallback, "clientCallback is null");
-        this.scheduler = requireNonNull(scheduler, "scheduler is null");
-        this.pageBufferClientCallbackExecutor = requireNonNull(pageBufferClientCallbackExecutor, "pageBufferClientCallbackExecutor is null");
-        requireNonNull(maxErrorDuration, "maxErrorDuration is null");
-        requireNonNull(ticker, "ticker is null");
-        this.backoff = new Backoff(maxErrorDuration, ticker);
-    }
-
-    public synchronized PageBufferClientStatus getStatus()
-    {
-        String state;
-        if (closed) {
-            state = "closed";
-        }
-        else if (future != null) {
-            state = "running";
-        }
-        else if (scheduled) {
-            state = "scheduled";
-        }
-        else if (completed) {
-            state = "completed";
-        }
-        else {
-            state = "queued";
-        }
-        String httpRequestState = "not scheduled";
-        if (future != null) {
-            httpRequestState = future.getState();
-        }
-
-        long rejectedRows = rowsRejected.get();
-        int rejectedPages = pagesRejected.get();
-
-        return new PageBufferClientStatus(
-                location,
-                state,
-                lastUpdate,
-                rowsReceived.get(),
-                pagesReceived.get(),
-                rejectedRows == 0 ? OptionalLong.empty() : OptionalLong.of(rejectedRows),
-                rejectedPages == 0 ? OptionalInt.empty() : OptionalInt.of(rejectedPages),
-                requestsScheduled.get(),
-                requestsCompleted.get(),
-                requestsFailed.get(),
-                httpRequestState);
-    }
-
-    public synchronized boolean isRunning()
-    {
-        return future != null;
     }
 
     @Override
-    public void close()
-    {
-        boolean shouldSendDelete;
-        Future<?> future;
-        synchronized (this) {
-            shouldSendDelete = !closed;
-
-            closed = true;
-
-            future = this.future;
-
-            this.future = null;
-
-            lastUpdate = DateTime.now();
-        }
-
-        if (future != null && !future.isDone()) {
-            future.cancel(true);
-        }
-
-        // abort the output buffer on the remote node; response of delete is ignored
-        if (shouldSendDelete) {
-            sendDelete();
-        }
-    }
-
-    public synchronized void scheduleRequest(DataSize maxResponseSize)
+    public ListenableFuture<PagesResponse> getResults(long token, DataSize maxResponseSize)
```

Comment:
I validated this implementation is refactored from 

https://github.com/prestodb/presto/blob/479e6552eff9f6e1776e3f40913d053e934241c5/presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java#L293-L300